### PR TITLE
BUG: Fixed timedelta numeric operations

### PR DESCRIPTION
**Pull request by TomAugspurger**
Closes https://github.com/pandas-dev/pandas/issues/18880

### CODE REVIEW

File: pandas/tests/test_nanops.py [hunk 1]
```diff
@@ -888,6 +888,19 @@ def test_nanstd_roundoff(self):
             result = data.std(ddof=ddof)
             assert result == 0.0
 
+    def test_nanvar_timedelta(self):
```
Comment (TomAugspurger) [inline pandas/tests/test_nanops.py]: I know this *should* be done up in `TestnanopsDataFrame` but I cannot figure out what's going on in those tests. `test_nanstd` and `test_nanvar` both have `allow_tdelta=True`, but were not catching the bug in https://github.com/pandas-dev/pandas/issues/18880

File: doc/source/whatsnew/v0.22.0.txt [hunk 2]
```diff
@@ -338,7 +338,8 @@ Reshaping
 Numeric
 ^^^^^^^
 
--
+- Fixed ``std`` and ``var`` computations for timedelta arrays not returning results in timedelta units (:issue:`18880`)
```
Comment (jreback) [inline doc/source/whatsnew/v0.22.0.txt]: this is a fix for 0.22? or for 0.23? (IOW do you need to have this for nansum)?

**Comment by TomAugspurger** [comment]
Doesn't seem like this is going to work without causing overflows everywhere...

**Review by jreback — CHANGES_REQUESTED** [review]

File: doc/source/whatsnew/v0.22.0.txt (hunk 2 above)
Comment (TomAugspurger) [reply doc/source/whatsnew/v0.22.0.txt]: This was was specific to nanstd, so I think I can work around it. The skipna stuff affected all the ops I think. I'll see how hard that is to work around.

Comment (jreback) [reply doc/source/whatsnew/v0.22.0.txt]: yeah rather leave this alone for now (IOW do it in 0.23).

Comment (jreback) [reply doc/source/whatsnew/v0.22.0.txt]: rebase on master. can you move to 0.23 (docs were renamed), prob easiest to just check this file from master and past in new one

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
can you move the release note and update

**Comment by jreback** [comment]
can you rebase

**Comment by jreback** [comment]
needs a rebase

**Pull request closed without merging**